lib/ansible/utils/__init__.py

```diff
@@ -1617,7 +1617,9 @@ def _load_vars_from_folder(folder_path, results, vault_password=None):
     names.sort()
 
     # do not parse hidden files or dirs, e.g. .svn/
-    paths = [os.path.join(folder_path, name) for name in names if not name.startswith('.')]
+    paths = [os.path.join(folder_path, name) for name in names
+            if not name.startswith('.')
+            and os.path.splitext(name)[1] in C.YAML_FILENAME_EXTENSIONS]
```

Comment:
This is `os.path.splitext` which always returns 2 items. So [1] and [-1] would always be the same.

https://docs.python.org/2/library/os.path.html#os.path.splitext
Split the pathname path into a pair (root, ext) such that root + ext == path, and ext is empty or begins with a period and contains at most one period. Leading periods on the basename are ignored; splitext('.cshrc') returns ('.cshrc', '').

```
In [1]: import os

In [2]: os.path.splitext('names.their.files.yml')
Out[2]: ('names.their.files', '.yml')
```